### ISSUE TITLE
Beta revision number not available for title

### ISSUE THREAD

**Issue by benplumley** [mozilla/nightlytt#204]
On the beta channel, the version number (as shown in About Firefox) is eg 47.0b2. However, none of NTT's variables for window name show the "b2" part. They only show "47.0".

**Comment by whimboo** [comment]
Interesting. We are getting the version via `this.appInfo.version`. And that clearly has not the beta version suffix. An additional check for the about window source shows me the following:

http://mxr.mozilla.org/mozilla-central/source/browser/base/content/aboutDialog.xul#48

`__MOZ_APP_VERSION_DISPLAY__` seems to be a placeholder which gets replaced during the build process of Firefox.

Because of that and no other instance of the beta revision in the application package, I'm not sure how to find out the beta index.

**Comment by benplumley** [comment]
The General Data section of about:telemetry shows the variable application.displayVersion with value 47.0b3. I don't know if that's something that can be queried.

**Comment by xabolcs** [comment]
Hi!

Telemetry's `application.displayVersion` traces into `TelemetryController.jsm`'s [`Impl._getApplicationSection()#384`][1]:
```
379     return {
380       architecture: arch,
381       buildId: Services.appinfo.appBuildID,
382       name: Services.appinfo.name,
383       version: Services.appinfo.version,
384       displayVersion: AppConstants.MOZ_APP_VERSION_DISPLAY,
385       vendor: Services.appinfo.vendor,
386       platformVersion: Services.appinfo.platformVersion,
387       xpcomAbi: Services.appinfo.XPCOMABI,
388       channel: updateChannel,
389     };
```
... where [`Appconstants`][2] can be imported with a `Cu.import`:
```
Cu.import("resource://gre/modules/AppConstants.jsm");
```

`AppConstants.jsm` was brought to you by [Bug 1124503][3] with milestone `mozilla39`.



It's OK to me to implement with `AppConstants.jsm` and fail silently if that isn't available.
Or put a `"displayVersion is not available in this version"` warning in the variable value.


[1]: http://mxr.mozilla.org/mozilla-central/source/toolkit/components/telemetry/TelemetryController.jsm#384
[2]: http://mxr.mozilla.org/mozilla-central/source/toolkit/modules/AppConstants.jsm
[3]: https://bugzilla.mozilla.org/show_bug.cgi?id=1124503 "Bug 1124503 - move AppConstants.jsm to toolkit"

**Comment by xabolcs** [comment]
@benplumley, thanks for the Telemetry hint! :+1:

**Comment by whimboo** [comment]
Great to see that AppConstants.jsm module. It would indeed be great to get it added.

**Comment by whimboo** [comment]
@xabolcs, this looks like an easy thing to get added now that we have experience with AppConstants.jsm. Do you think we could fold this into the 3.9 release?

**Comment by xabolcs** [comment]
Sure we could, but please see the conversation in PR #212, starting with [my comment][1]!


[1]: https://github.com/mozilla/nightlytt/pull/212#issuecomment-243001998

**Comment by UtiluMark** [comment]
I'm currently implementing the added customize title bar options into the WebExtension.

In the XUL-based extension the beta version was retreived using:
```
Components.utils.import("resource://gre/modules/AppConstants.jsm");
AppConstants.MOZ_APP_VERSION_DISPLAY;
```
In Firefox 57
resource://gre/modules/AppConstants.jsm
still contains the beta version number, just like in earlier versions:
`MOZ_APP_VERSION_DISPLAY: "57.0b7",`
but retreiving this using the code from the XUL-based extension doesn't work in the WebExtension.

@kyoshino Do you know a way to retreive the beta version in the WebExtension?

**Comment by xabolcs** [comment]
@UtiluMark, try with [WebExtensions Experiments](https://webextensions-experiments.readthedocs.io/en/latest/)!

**Comment by whimboo** [comment]
Yes it is. And checking again it should basically be fixed at the WebExtension API level. There is https://bugzilla.mozilla.org/show_bug.cgi?id=1389751 open for exactly that problem.
